src/daemon.py

```diff
@@ -30,100 +29,74 @@ def compile_yara(job_hash: str) -> Any:
     return rule
 
 
-def get_list_name(priority: str) -> str:
-    if priority == "low":
-        return "list-yara-low"
-    elif priority == "medium":
-        return "list-yara-medium"
-    else:
-        return "list-yara-high"
-
-
 def collect_expired_jobs() -> None:
     if config.JOB_EXPIRATION_MINUTES <= 0:
         return
 
     exp_time = int(60 * config.JOB_EXPIRATION_MINUTES)  # conversion to seconds
-    job_hashes = []
+    job_ids: List[JobId] = []
 
-    for job_hash in redis.keys("job:*"):
-        job_hashes.append(job_hash[4:])
+    for job in redis.get_job_ids():
+        job_ids.append(job)
 
-    for job in job_hashes:
-        redis.set("gc-lock", "locked", ex=60)
-        job_submitted_time = int(redis.hget("job:" + job, "submitted"))
+    for job in job_ids:
+        job_submitted_time = redis.get_job_submitted(job)
         if (int(time.time()) - job_submitted_time) >= exp_time:
-            redis.hset("job:{}".format(job), "status", "expired")
-            redis.delete("meta:{}".format(job))
+            redis.expire_job(job)
 
 
 def process_task(queue: str, data: str) -> None:
     if queue == "queue-search":
-        job_hash = data
-        logging.info(f"New task: {queue}:{job_hash}")
+        job = JobId(data)
+        logging.info(f"New task: {queue}:{job.hash}")
 
         try:
-            execute_search(job_hash)
+            execute_search(job)
         except Exception as e:
             logging.exception("Failed to execute job.")
-            redis.hmset(
-                "job:" + job_hash, {"status": "failed", "error": str(e)}
-            )
+            redis.fail_job(None, job, str(e))
     elif queue == "queue-commands":
         logging.info("Running a command: %s", data)
         resp = db.execute_command(data)
         logging.info(resp)
 
 
-def get_random_job_by_priority() -> Tuple[Optional[str], str]:
-    yara_lists = ["list-yara-high", "list-yara-medium", "list-yara-low"]
-    for yara_list in yara_lists:
-        yara_jobs = redis.lrange(yara_list, 0, -1)
-        if yara_jobs:
-            return yara_list, random.choice(yara_jobs)
-    return None, ""
-
-
 def try_to_do_task() -> bool:
-    task_queues = ["queue-search", "queue-commands"]
-    task = None
-    for queue in task_queues:
-        task = redis.lpop(queue)
-        if task is not None:
-            data = task
-            process_task(queue, data)
-            return True
+    queue, task = redis.get_task()
+    if task is not None:
+        data = task
+        process_task(str(queue), str(data))
```

Comment:
Just making sure, are the `str`s are here on purpose?
`get_task()` might return None values so process_task might be ran like `process_task("None", something)`

Reply:
Actually it can't, because it will either return `None, None` or `str, str`. I've refactored this like it was suggested elsewhere with the other suggestion, so they are not necessary anymore.

Reply:
Oh so the tostring methods were added just to stop the linter from complaining? 😈

> Actually it can't, because it will either return None, None or str, str

Not sure if this is the correct attitude, if type hints tell us that `None, str` might be returned, shouldn't we be prepared for handling it?
This kind of assumption seems very bug-inducing :thinking:

Reply:
The refactor lgtm tho

Reply:
`Oh so the tostring methods were added just to stop the linter from complaining? smiling_imp` well actually yes (because it technically was a correct code, it was linter that was wrong).

```
if type hints tell us that None, str might be returned, shouldn't we be prepared for handling it?
```
I guess the type should be `Union[Tuple[str, str], Tuple[None, None]` in the previous case :thinking:

Reply:
I mean, I agree that we shouldn't assume more than types say using knowledge about function internals, so good catch.